modules/graphql_core/tests/modules/graphql_batched_test/src/Plugin/GraphQL/Fields/Users.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\graphql_batched_test\Plugin\GraphQL\Fields;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\graphql_batched_test\UserDataBaseInterface;
+use Drupal\graphql_core\BatchedFieldResolver;
+use Drupal\graphql_core\GraphQL\BatchedFieldInterface;
+use Drupal\graphql_core\GraphQL\FieldPluginBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Youshido\GraphQL\Execution\ResolveInfo;
+
+/**
+ * A basic user field.
+ *
+ * @GraphQLField(
+ *   id = "users",
+ *   name = "users",
+ *   type = "User",
+ *   multi = true,
+ *   arguments = {
+ *     "uids" = {
+ *       "type" = "String",
+ *       "multi" = true
+ *     }
+ *   }
+ * )
+ */
+class Users extends FieldPluginBase implements ContainerFactoryPluginInterface, BatchedFieldInterface {
+
+  /**
+   * The user database.
+   *
+   * @var \Drupal\graphql_batched_test\UserDataBaseInterface
+   */
+  protected $userDatabase;
+
+  /**
+   * The batched field resolver.
+   *
+   * @var \Drupal\graphql_core\BatchedFieldResolver
+   */
+  protected $batchedFieldResolver;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getBatchId($parent, array $arguments) {
+    return "users";
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getBatchedFieldResolver() {
+    return $this->batchedFieldResolver;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareBatch(array $batch) {
```

Comment:
It fetches the data and splits it up... So actually it does more than just preparation?